### PR TITLE
[MINOR] Update javadoc for RexChecker with Litmus argument

### PR DESCRIPTION
**Pull request by yeahthisisrob**
Update the javadoc for `RexChecker` to reflect changes from https://issues.apache.org/jira/browse/CALCITE-955

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rex/RexChecker.java [hunk 1]
```diff
@@ -31,31 +31,24 @@
...
+ * <li>{@link Litmus#IGNORE} tests validity without throwing an error.</li>
  * </ul>
  *
+ * <p>Example usage:
+ * <blockquote><pre>
+ * RexNode node;
+ * RelDataType rowType;
+ * RexChecker checker = new RexChecker(rowType, Litmus.IGNORE);
+ * node.accept(checker);
+ * if (!checker.isValid()) {
```
Comment (mihaibudiu) [inline core/src/main/java/org/apache/calcite/rex/RexChecker.java:47]: The original javadoc had two examples, whereas here there is only one.
I do find that the assert-based example is useful.

**Comment by julianhyde** [comment]
Remove 'Fix:' from the commit message. That's not a prefix we use in this project.

File: core/src/main/java/org/apache/calcite/rex/RexChecker.java (hunk 1 above)
Comment (yeahthisisrob) [reply core/src/main/java/org/apache/calcite/rex/RexChecker.java:53]: Thanks, I completely missed the first example. I recently used `IGNORE` and was focused there. Updated!

**Review by mihaibudiu — APPROVED** [review]

**Comment by JiajunBernoulli** [comment]
@yeahthisisrob , Would you please add `[MINOR]` to commit message？

**Pull request merged**